ext/SummationByPartsOperatorsOptimForwardDiffExt.jl

```diff
@@ -0,0 +1,222 @@
+module SummationByPartsOperatorsOptimForwardDiffExt
+
+using Optim: Optim, Options, LBFGS, optimize, minimizer
+using ForwardDiff: ForwardDiff
+
+using SummationByPartsOperators: SummationByPartsOperators, GlaubitzNordströmÖffner2023, MatrixDerivativeOperator
+using LinearAlgebra: Diagonal, LowerTriangular, dot, diag, norm, mul!
+using PreallocationTools: DiffCache, get_tmp
```

Comment:
Are these still required after implementing the gradient analytically?

Reply:
Ah no, I don't think so since I introduced them to be able to apply automatic differentiation. I'll remove PreallocationTools again.

Reply:
I removed it again in [7f54a30](https://github.com/ranocha/SummationByPartsOperators.jl/pull/268/commits/7f54a303cfcb0c0f5be49700223c546de6ce83e3).